api/spec/spec_handler.go

```diff
@@ -472,7 +473,15 @@ func (d *specHandler) UpdateSpecFromOpts(opts map[string]string, spec *api.Volum
 				spec.Sharedv4ServiceSpec = &api.Sharedv4ServiceSpec{}
 			}
 			spec.Sharedv4ServiceSpec.Name = v
-
+		case api.SpecSharedv4FailoverDelay:
+			if spec.Sharedv4Spec == nil {
+				spec.Sharedv4Spec = &api.Sharedv4Spec{}
+			}
+			if v == "short" {
+				spec.Sharedv4Spec.FailoverDelay = api.Sharedv4FailoverDelay_SHORT
+			} else if v == "normal" {
+				spec.Sharedv4Spec.FailoverDelay = api.Sharedv4FailoverDelay_NORMAL
+			}
```

Comment:
Add an else here to error out for all unknown strings.

Also add the short and normal strings as constants